### ISSUE TITLE
Increase or document environment branch reference limits

### ISSUE THREAD

**Issue by shreddedbacon** [uselagoon/lagoon#3815]
The baseref and headref fields on an environment in the database have a max length of 100. This can result in environments or pullrequests that have long branch refs failing to deploy with an obscure error.

https://github.com/uselagoon/lagoon/blob/1cf5c769be03f5b4d17ab1d8ea0ed9faf2a25102/services/api/database/migrations/20220908065119_initial_db.js#L79-L80

We should either increase this, or document these limits. We should also return an clearer error if these limits are exceeded, rather than when the SQL in performed.

**Comment by tobybellwood** [comment]
It looks like github has a 255 char limit - including `refs/heads/`

so maybe we can change this - but should we also support longer environment names too - to accommodate it - or are we happy to truncate them when needed?

Context - making lots of db changes in https://github.com/uselagoon/lagoon/pull/3816 - so can accommodate these there if needed

**Comment by shreddedbacon** [comment]
> so maybe we can change this - but should we also support longer environment names too - to accommodate it - or are we happy to truncate them when needed?

TLDR; Yeah, we can probably increase them to 255, but leave the environment name at 100.

Well we add some things into [labels in resources](https://github.com/uselagoon/build-deploy-tool/blob/7ad87f8c3248effcbd30f34997f2f64724141cee/internal/templating/services/templates_deployment.go#L41-L42), and labels in kubernetes have limits on the [length of the value](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L159). If we were going to have any issues, it would be on the `environment name` after it has [been modified](https://github.com/uselagoon/lagoon/blob/daaf1b3cd107295bc1b0686cdfae4561ae5f3d3b/node-packages/commons/src/tasks.ts#L366-L376), but this truncates the calculated name to 63 already (58+5, 5 being `-hash`).

Luckily, we put the unmodified values of branch/refs [into annotations](https://github.com/uselagoon/build-deploy-tool/blob/7ad87f8c3248effcbd30f34997f2f64724141cee/internal/templating/services/templates_deployment.go#L58-L63) which can definitely support more than 255.

I don't think we do anything to the project name though, and this would suffer from a being greater than 63 thing, which I'm a bit surprised hasn't happened. We limit project names to [100 in the db](https://github.com/uselagoon/lagoon/blob/daaf1b3cd107295bc1b0686cdfae4561ae5f3d3b/services/api/database/migrations/20220908065119_initial_db.js#L208)

**Comment by shreddedbacon** [comment]
Closed by https://github.com/uselagoon/lagoon/pull/3816 which increases these to 250 characters.
